lib/fog/ovirt/compute/v4.rb

```diff
@@ -167,13 +166,21 @@ def datacenter
           end
 
           def datacenter_hash
+            @datacenter_hash ||= datacenters.find { |x| x[:id] == datacenter } if datacenter
             @datacenter_hash ||= datacenters.first
           end
 
           def blank_template
             @blank_template ||= client.system_service.get.special_objects.blank_template
           end
 
+          def create_search_by_datacenter(search, page)
```

Comment:
If you don't always have values for page or datacenter name, you can make them [named parameters](https://robots.thoughtbot.com/ruby-2-keyword-arguments), or at least put defaults, so you will not have to pass `nil` if you don't have something to put there.